Application/EdFi.Ods.Api/Security/Authorization/AuthorizationSegmentSqlProviderBase.cs

```diff
@@ -45,100 +46,80 @@ public AuthorizationSegmentSqlProviderBase(IAuthorizationViewsProvider authoriza
         {
             var segmentStatements = new List<string>();
             var parameters = new List<DbParameter>();
+            var unsupportedAuthorizationViews = new List<string>();
+            var authorizationPathModifiers = new List<string>();
 
             foreach (var authorizationSegment in authorizationSegments)
             {
-                // Within each claim segment, group the values by ed org type, and combine with "OR"
-                var claimEndpointsGroupedByName = authorizationSegment.ClaimsEndpoints
-                    .GroupBy(ep => ep.Name)
-                    .Select(g => g)
-                    .ToList();
+                if(!EducationOrganizationEntitySpecification.IsEducationOrganizationIdentifier(authorizationSegment.ClaimsEndpoints.FirstOrDefault().Name))
```

Comment:
There's a few issues with how this is written:
- It's not only the first claim that we care about, it's all of them.
- I am pretty sure that, as implemented, the same claim values will be present on all segments so the defensive check should be written to check for `Any` claims that are not EdOrgs, but it should only do this on the first segment.
- The way you wrote this, if `FirstOrDefault()` actually returns a `null`, then the subsequent `.Name` will throw an unhandled null reference exception -- very undesirable. If you used the `First()` instead, you'd get a slightly less obtuse exception message from Linq, but it's still undesirable. Try to avoid writing code with these sorts of possible outcomes by adding guard checks for the error conditions and throw custom exceptions that explain the problem to poor developer who might run into it later. In this particular case, I don't think this code execution would get this far without claims assigned, but it's better to deal with it explicitly and clearly.